# Introduction
Staphylococcus aureus is a commensal and major infectious pathogen of humans and animals. In humans, 25% of the population are persistent asymptomatic carriers in the anterior nares of the nose, and a further 50% are transient carriers (von Eiff et al., 2001, Safdar and Bradley, 2008). S. aureus can cause a diverse range of infections from minor skin and soft tissue infections (SSTI) to more severe, and often fatal, systemic infections. Invasive infections are often caused by a patients’ own colonising strain (McCarthy et al., 2012a) indicating that the same previously harmless bacteria have the potential to cause a wide range of opportunistic infections. The emergence of methicillin-resistant S. aureus (MRSA) is concerning as these reservoirs are resistant to β-lactamase resistant β-lactam antibiotics, the preferred treatment for staphylococcal infections, and/or other commonly prescribed antibiotics (Gould, 2005, Knight et al., 2012).

Evasion of host immune responses is crucial for pathogen survival in the host. The innate immune response provides the primary defence against S. aureus infections, and is highly dependent on complement activation and neutrophil-mediated killing (Fig. 1) (Rigby and DeLeo, 2012). Indeed, neutrophil defects are often associated with increased susceptibility to S. aureus infections (Miller and Cho, 2011). Protective immunity against S. aureus is not observed and recurrent infections frequently occur (Kim et al., 2012); this suggests adaptive immune responses are not as important as innate immune responses for S. aureus clearance.

Genetic variants of S. aureus immune evasion proteins have been described, and subsequent biochemical studies have shown that variants of the same gene may function differently. The best-characterised example is the finding that the Flipr/Flipr-like protein from S. aureus lineage clonal complex (CC)1 binds formyl peptide receptor-1 (FPRL-1) and formyl peptide receptor (FPR), whilst the equivalent protein from S. aureus lineage CC5 only binds to FPR (Prat et al., 2006, Prat et al., 2009). This suggests that there could be differences in the function and strategies of immune evasion in different lineages. All S. aureus isolates belong to a lineage (Lindsay et al., 2006). Each lineage is evolving independently and carries a unique but highly conserved combination of “core variable” genes that include genes encoding surface protein complexes, secreted immune evasion protein complexes and their regulators (McCarthy and Lindsay, 2010, Lindsay et al., 2006). Nucleotide variation in core variable genes is often non-synonymous resulting in production of highly variable proteins. In contrast, there is a lower magnitude of variation in house-keeping genes, although this is sufficient for identifying clonal lineage by multi-locus sequence typing (MLST). The majority of human isolates belong to lineages CC1, CC5, CC8, CC9, CC12, CC15, CC22, CC30, CC45 and CC51 (Feil et al., 2003).

Additional S. aureus innate immune evasion protein complexes are often encoded on mobile genetic elements (MGEs). MGEs include bacteriophage, S. aureus pathogenicity islands (SaPIs), plasmids, transposons and staphylococcal chromosome cassettes (SCCs) (Lindsay and Holden, 2004). MGEs are pieces of DNA that can move between bacteria via horizontal gene transfer (HGT) mechanisms. MGE distributions are often lineage associated because each lineage has a unique Sau1 restriction-modification (RM) system that limits HGT between lineages (Waldron and Lindsay, 2006, McCarthy et al., 2012b, McCarthy et al., 2012c,). MGE distributions are also associated with host and geographic origin (McCarthy et al., 2011, McCarthy and Lindsay, 2012). The first well-characterised MGE-encoded immune evasion protein complexes are the human-specific immune evasion cluster (IEC) protein complexes chemotaxis inhibitory protein (Chp), Staphylococcal complement inhibitor (Scn) and Staphylokinase (Sak). The genes encoding these protein complexes are located on φ3 bacteriophage (van Wamel et al., 2006) that are common amongst human, but not animal, S. aureus isolates (Sung et al., 2008). The second well-characterised example is the human-specific Panton-Valentine leukocidin (PVL) genes lukFS-PVL. These genes are located on φ2 bacteriophage (Kaneko et al., 1998, Löffler et al., 2010), and are common amongst invasive community-associated (CA)-MRSA isolates (Vandenesch et al., 2003). Currently, there is a poor understanding of animal-specific S. aureus immune evasion protein complexes. A homolog of the scn gene has recently been identified in ruminant isolates of lineage CC133 (Guinane et al., 2010), but no function has yet been described. Knowledge of host-specific immune evasion protein complexes could identify novel opportunities for the control of S. aureus populations in different host environments.

The overall aim of this bioinformatic study was to systematically compare the diversity of secreted immune evasion protein complexes in the sequenced S. aureus population. Specifically, the study aimed to identify genes that vary and could be investigated for different immunological functions, and to identify which protein complexes have lineage-specific and host-specific distributions.